### PR TITLE
Get the vector from one point to another expressible in a single reference frame

### PR DESCRIPTION
**Pull request by shattar**
Added the ability to get the vector from one point to another that is expressed, or expressible, in a single reference frame.

#### Brief description of what is fixed or changed
Added options to the `pos_from` method of the `Point` class in order to specify that the resultant vector should be expressible in the given reference frame, or any reference frame at all.

This is needed because the pos_from method was previously creating a vector along the shortest number of connected points, even if there was no orientation specified between the reference frames through which those relative point locations were defined.  The new options will allow the selection of the smallest chain of connected points whose resultant vector is expressible in a single frame.

#### Other comments
A new method was added to the `Vector` class called `check_expressible` in order avoid having to go through the entire dcm transformation of the vector just to find out if the connection exists.

#### Release Notes

* physics.vector
  * `Point.pos_from` method is now able to return the resultant vector in a specified reference frame, if provided.

### CODE REVIEW

**Comment by shattar** [comment]
Some tests are failing due to stack overflow during test execution.  This is not something introduced with this pull request.

> sympy/combinatorics/coset_table.py[3] . Fatal Python error: Cannot recover from stack overflow.

**Comment by moorepants** [comment]
I have to get through grading this week and can have a look come the weekend or next week. Thanks for the addition! Seems like a good idea.

**Comment by asmeurer** [comment]
I opened https://github.com/sympy/sympy/issues/13737 for the test failure. It's probably intermittent, meaning it will likely go away once you push another commit.

**Comment by ylemkimon** [comment]
Please rebase on sympy/master. Thank you!

File: sympy/physics/vector/point.py [hunk 1]
```diff
@@ -215,14 +219,24 @@ def locatenew(self, name, value):
...
+    def pos_from(self, otherpoint, frame=None, variables=False, dcm_connected=False):
         """Returns a Vector distance between this Point and the other Point.
 
         Parameters
         ==========
 
         otherpoint : Point
             The otherpoint we are locating this one relative to
+        frame : ReferenceFrame
+            Expresses the resulting vector in this frame
```
Comment (moorepants) [inline sympy/physics/vector/point.py]: Initial reaction is that we should leave this to the `express()` method on `Vector`. Why is it necessary to have this here?

File: sympy/physics/vector/tests/test_point.py [hunk 2]
```diff
@@ -112,6 +112,19 @@ def test_point_pos():
...
+    N = ReferenceFrame('N')
+    B = ReferenceFrame('B')
+    C = ReferenceFrame('C')
+    O = Point('O')
+    P = O.locatenew('P', 10 * N.x + 5 * B.x)
+    raises(ValueError, lambda: P.pos_from(O, dcm_connected=True))
+    B.orient(N, 'Axis', [q, N.z])
+    assert P.pos_from(O, dcm_connected=True) == 10 * N.x + 5 * B.x
+    assert P.pos_from(O, frame=N) == 10 * N.x + 5 * B.x
+    assert P.pos_from(O, frame=B) == 10 * N.x + 5 * B.x
```
Comment (moorepants) [inline sympy/physics/vector/tests/test_point.py]: Why is `P.pos_from(O).express(N)` not possible?

Comment (moorepants) [reply sympy/physics/vector/tests/test_point.py]: Also, why would there be B unit vectors if you set `frame=N)?

**Review by moorepants — CHANGES_REQUESTED** [review]
I don't quite understand what this is doing? If you create a new position with `locatenew()`, why doesn't `pos_from()` just return that vector?

**Comment by moorepants** [comment]
Needs release notes as per: https://github.com/sympy/sympy/wiki/Writing-Release-Notes

**Comment by czgdp1807** [comment]
@moorepants I have updated the release notes entry. Please take a look at this PR.

**Comment by moorepants** [comment]
I raised several issues two years ago about this, specifically regarding it be a redundancy that `express()` already handles fine. As it stands I do not support merging this.

**Comment by czgdp1807** [comment]
So probably we should close this PR.

**Comment by czgdp1807** [comment]
Closing this PR for now as the reviews are not supportive of merging. However, please let us know if there is a need to re-open this. Thanks for your contributions.

**Pull request closed without merging**